crates/uv-client/src/registry_client.rs

```diff
@@ -78,11 +79,26 @@ impl RegistryClientBuilder {
 
     pub fn build(self) -> RegistryClient {
         let client_raw = {
+            // Get pip timeout from env var
+            let default_timeout = 5 * 60;
+            let timeout: u64 = match env::var("UV_REQUEST_TIMEOUT") {
+                Ok(value) => {
+                    match value.parse::<u64>() {
+                        Ok(parsed) => parsed,
+                        Err(_) => {
+                            eprintln!("Warning: UV_REQUEST_TIMEOUT is an invalid value. Using default value.");
+                            default_timeout
+                        }
+                    }
+                }
+                Err(_) => default_timeout,
+            };
+            debug!("Pip request timeout is {}.", timeout);
```

Comment:
We should probably not say "Pip" this is our general registry client 
```suggestion
            debug!("Using registry request timeout of {}s", timeout);
```